modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.shared/src/main/scala/edu/gemini/seqexec/web/common/FixedLengthBuffer.scala

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package edu.gemini.web.common
+
+import scalaz.{Show, Equal, Functor}
+import scalaz.syntax.std.boolean._
+
+object FixedLengthBuffer {
+  private final case class FixedLengthBufferImpl[A](maxLength: Int, data: Vector[A]) extends FixedLengthBuffer[A] {
+    // Sanity check
+    require(maxLength >= data.length)
+    require(maxLength >= 0)
+
+    def append(element: A): FixedLengthBuffer[A] = {
+      if (data.length == maxLength && data.length >= 0) {
+        FixedLengthBufferImpl[A](maxLength, data.tail :+ element)
```

Comment:
So this means when maximum buffer length is reached the first messages are deleted, right? Nice 👍

Reply:
Yes we keep only N elements to avoid killing the memory on the client